From 6ce5dc3d0a58326fb42f7065d387f0238ebb315a Mon Sep 17 00:00:00 2001
From: Ding Qiang <b02550@freescale.com>
Date: Fri, 19 Jun 2009 18:37:55 +0800
Subject: [PATCH 102/449] ENGR00113242 IPUv3: Fix the variable parameters
 setting issue.

The varialbe parameters should be set in the unit test, and the default
value should not be trust.

Signed-off-by: DingQiang <b02550@freescale.com>
---
 test/mxc_ipudev_test/test_pattern.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/test/mxc_ipudev_test/test_pattern.c b/test/mxc_ipudev_test/test_pattern.c
index 9724ec2..1975323 100644
--- a/test/mxc_ipudev_test/test_pattern.c
+++ b/test/mxc_ipudev_test/test_pattern.c
@@ -362,6 +362,16 @@ int color_bar(int two_output, int overlay, ipu_test_handle_t * test_handle)
 		goto done;
 	}
 
+	if(fb_var.yres_virtual != 2*fb_var.yres)
+	{
+		fb_var.yres_virtual = 2*fb_var.yres;
+		if ( ioctl(fd_fb, FBIOPUT_VSCREENINFO, &fb_var) < 0) {
+			printf("Get FB var info failed!\n");
+			ret = -1;
+			goto done;
+		}
+	}
+
 	screen_size = fb_var.yres * fb_fix.line_length;
 	fb[0] = mmap(NULL, screen_size, PROT_READ | PROT_WRITE, MAP_SHARED,
 			fd_fb, 0);
@@ -877,7 +887,7 @@ void * thread_func_color_bar(void *arg)
 	memset(&test_handle, 0, sizeof(ipu_test_handle_t));
 
 	test_handle.ipu_handle = &ipu_handle;
-	color_bar(0, NO_OV, &test_handle);
+	ret = color_bar(0, NO_OV, &test_handle);
 
 	pthread_exit((void*)ret);
 }
-- 
1.8.0

